fatiando/gridder.py

```diff
@@ -439,3 +444,269 @@ def cut(x, y, scalars, area):
+
+    Returns:
+
+    * ap : numpy array
+        Padded array. The array core is a deep copy of the original array
+    * cp : list
+        List of coordinate arrays containing the extrapolated coordinate values
+        Returns None if xy not provided
+    * nps : list
+        List of tuples containing the number of elements padded onto each 
+        dimension.
+
+    """
+
+    # Test to make sure padtype is valid
+    padopts = ['','oddreflectiontaper','oddreflection', 'reflection',
+            'lintaper', 'edge', 'value', 'mean']
+    #API Note:
+    # If one wishes to add more options to the padding, use the following
+    # checklist to make sure all the sections are consistent.
+    #  [ ] Docstring (2 spots: the options, and the option descriptions
+    #  [ ] padopts - Make sure to add to avoid raising a ValueError
+    #  [ ] if statements below - Finally add an elif to the if statements
+    #                            below. Add a descriptive comment.
+    if padtype.lower() not in padopts and not _is_number(padtype):
+        raise ValueError('gridder.pad_array: Pad Type not understood')
+    # If npd is not provided, populate with next power of 2
+    npt = []
+    nd = a.ndim
+    if npd == None:
+        for ii in range(0, nd):
+            if nd == 1:
+                npt.append(_nextpow2(len(a)))
+            else:
+                npt.append(_nextpow2(a.shape[ii]))
+    else:
+        et = 'gridder.pad_array: pad dimensions do not match array dims'
+        if nd == 1:
+            if _is_integer(npd):
+                npt.append(npd)
+            else:
+                if len(npd) != 1:
+                    raise ValueError(et)
+                else:
+                    npt.append(npd)
+        else:
+            if _is_integer(npd):
+                raise ValueError(et)
+            elif len(npd) != a.ndim:
+                raise ValueError(et)
+            else:
+                npt = npd
+        for ii in numpy.arange(len(npt)):
+            if npt[ii] <= a.shape[ii]:
+                # How are you supposed to properly assign a multi-line string?
+                et1 = 'gridder.pad_array: Desired padding is less than array '
+                et2 = 'length along dimension '+str(ii)
+                et = et1+et2
+                raise ValueError(et)
+    # Compute numbers to pad on the left and right side of the array along
+    # each dimension
+    nps = []
+    for ii in range(0, nd):
+        nps.append((int(numpy.ceil((npt[ii] - a.shape[ii])/2.)),
+            int(numpy.floor((npt[ii] - a.shape[ii])/2.))))
+
+    # If it will be needed, compute the mean
+    meanneeded = ['lintaper', 'oddreflectiontaper']
+    if padtype.lower() in meanneeded:
+        m = numpy.mean(a)
+    # Use numpy's padding routines where possible
+    if _is_number(padtype):
+        # Pad with value
+        ap = numpy.pad(numpy.copy(a), nps, mode='constant',
+            constant_values = float(padtype))
```

Comment:
Doesn't `numpy.pad` already produce a copy? I would think so. Usually the inplace operations are done via array methods.